### PR TITLE
3248, fixed an issue with role composisition removal

### PR DESCRIPTION
**Pull request by tomdoischer**
https://redmine.czechidm.com/issues/3248

### CODE REVIEW

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 1]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
 			@PathVariable @NotNull String backendId) {
 		return super.getPermissions(backendId);
 	}
+
+	/**
+	 * Get available bulk actions for account
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:274]:
```suggestion
	 * Get available bulk actions for role composition
```

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 2]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
...
+							@AuthorizationScope(scope = CoreGroupPermission.ROLECOMPOSITION_READ, description = "") }),
+					@Authorization(value = SwaggerConfig.AUTHENTICATION_CIDMST, scopes = {
+							@AuthorizationScope(scope = CoreGroupPermission.ROLECOMPOSITION_READ, description = "") })
+			})
+	public List<IdmBulkActionDto> getAvailableBulkActions() {
+		return super.getAvailableBulkActions();
+	}
+
+	/**
+	 * Process bulk action for accounts
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:297]:
```suggestion
	 * Process bulk action for role compositions
```

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 3]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
...
+	 *
+	 * @param bulkAction
+	 * @return
+	 */
+	@Override
+	@ResponseBody
+	@RequestMapping(path = "/bulk/action", method = RequestMethod.POST)
+	@PreAuthorize("hasAuthority('" + CoreGroupPermission.ROLECOMPOSITION_READ + "')")
+	@ApiOperation(
+			value = "Process bulk action for account",
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:307]:
```suggestion
			value = "Process bulk action for role composition",
```

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 4]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
...
+							@AuthorizationScope(scope = CoreGroupPermission.ROLECOMPOSITION_READ, description = "")}),
+					@Authorization(value = SwaggerConfig.AUTHENTICATION_CIDMST, scopes = {
+							@AuthorizationScope(scope = CoreGroupPermission.ROLECOMPOSITION_READ, description = "")})
+			})
+	public ResponseEntity<IdmBulkActionDto> bulkAction(@Valid @RequestBody IdmBulkActionDto bulkAction) {
+		return super.bulkAction(bulkAction);
+	}
+
+	/**
+	 * Prevalidate bulk action for accounts
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:322]:
```suggestion
	 * Prevalidate bulk action for role compositions
```

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 5]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
...
+	 *
+	 * @param bulkAction
+	 * @return
+	 */
+	@Override
+	@ResponseBody
+	@RequestMapping(path = "/bulk/prevalidate", method = RequestMethod.POST)
+	@PreAuthorize("hasAuthority('" + CoreGroupPermission.ROLECOMPOSITION_READ + "')")
+	@ApiOperation(
+			value = "Prevalidate bulk action for accounts",
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:332]:
```suggestion
			value = "Prevalidate bulk action for role compositions",
```

File: Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java [hunk 6]
```diff
@@ -266,4 +269,96 @@ public Set<String> getPermissions(
...
+	public ResponseEntity<ResultModels> prevalidateBulkAction(@Valid @RequestBody IdmBulkActionDto bulkAction) {
+		return super.prevalidateBulkAction(bulkAction);
+	}
+
+	@Override
+	@ResponseBody
+	@RequestMapping(path = "/bulk/preprocess", method = RequestMethod.POST)
+	@PreAuthorize("hasAuthority('" + CoreGroupPermission.ROLECOMPOSITION_READ + "')")
+	@ApiOperation(
+			value = "Preprocess bulk action for accounts",
```
Comment (peter-strunc) [inline Realization/backend/core/core-impl/src/main/java/eu/bcvsolutions/idm/core/rest/impl/IdmRoleCompositionController.java:351]:
```suggestion
			value = "Preprocess bulk action for role compositions",
```

**Review by peter-strunc — CHANGES_REQUESTED** [review]
Looks good, working well. Only a few issues regarding javadoc. GJ!

**Comment by tomdoischer** [comment]
Javadoc in the rest controller updated, nice catch, thank you!

**Pull request merged**